source/MaterialXGenShader/UnitSystem.cpp

```diff
@@ -143,23 +143,33 @@ UnitSystemPtr UnitSystem::create(const string& language)
     return UnitSystemPtr(new UnitSystem(language));
 }
 
-string UnitSystem::getImplementationName(const UnitTransform& transform, const string& unitname) const
+ImplementationPtr UnitSystem::getImplementation(const UnitTransform& transform, const string& unitname) const
 {
-    return "IM_" + unitname + "_unit_" + transform.type->getName() + "_" + _target;
+    // Search up the targetdef derivation hierarchy for a matching implementation.
+    TargetDefPtr targetDef = _document->getTargetDef(_target);
```

Comment:
Use same logic as for DefaultColorManagementSystem::getImplementation() which was patched to search all targets in targetdef hierarchy.